### PR TITLE
Add buy and sell endpoints and tests

### PR DESCRIPTION
**Pull request by sophiajmoore**
# Overview

In UserCommonsController.java, endpoints for Buy and Sell buttons have been added. When the Buy endpoint is called, the Total Wealth decreases by Cow Price, and the number of cows increases by one. When the Sell endpoint is called, the Total Wealth increases by Cow Price, and the number of cows decreases by one. 

The UserCommonsControllerTests.java file now contains tests for these two endpoints and all backend tests pass. Tests include: buying/selling a cow when commons exists, buying/selling a cow when the commons for user does not exist, buying/selling a cow when the commons does not exist, and finally tests the edge cases of buying a cow when you do not have enough money (you will not be able to) or attempting to sell a cow when you have none (you will not be able to).

# Issues Addressed

This PR addresses issues #14 and #15. Also starts and completes issue #49.

# Details

The two below images show the Buy endpoint working.
Screen Shot 2022-03-13 at 8 50 31 PM
Screen Shot 2022-03-13 at 8 50 37 PM

The two below images show the Sell endpoint working.
Screen Shot 2022-03-13 at 8 51 02 PM
Screen Shot 2022-03-13 at 8 51 05 PM

# Testing Plan
1) Create and join a commons.
2) On swagger-ui, go to the buy endpoint. Fill out the ids. Then, execute and notice that on the webpage, the total wealth and number of cows changes accordingly. Do the same for the sell endpoint.
3) Note that if you attempt to sell a cow but have no cows you will be unable to, and that you will be unable to buy a cow if your total wealth is $0.

### CODE REVIEW

**Review by pconrad — APPROVED** [review]
LGTM!

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java [hunk 1]
```diff
@@ -57,4 +69,59 @@ public UserCommons getUserCommonsById(
...
+          @ApiParam("commonsId") @RequestParam Long commonsId, 
+            @RequestBody @Valid UserCommons incomingUserCommons) throws JsonProcessingException {
+        User u = getCurrentUser().getUser();
+        Long userId = u.getId();
+
+        Commons commons = commonsRepository.findById(commonsId).orElseThrow( 
+          ()->new EntityNotFoundException(Commons.class, commonsId));
+        UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
+        .orElseThrow(
+            () -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java]: Instead of passing in a UserCommons object as the "incomingUserCommons", it would make more sense to just look up that using the commonsId, and the userId that you get from:
```
User u = getCurrentUser().getUser();
```

So when you do 

```
UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
```

the userId would be u.getId():

And you'd remove the `@RequestBody @Valid UserCommons incomingUserCommons`

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java [hunk 2]
```diff
@@ -57,4 +69,59 @@ public UserCommons getUserCommonsById(
...
+
+        
+        long previousId = userCommons.getId();
+        incomingUserCommons.setId(previousId);
+        if(incomingUserCommons.getTotalWealth() >= commons.getCowPrice() ){
+          incomingUserCommons.setTotalWealth(incomingUserCommons.getTotalWealth() - commons.getCowPrice());
+          incomingUserCommons.setNumOfCows(incomingUserCommons.getNumOfCows() + 1);
+        }
+        userCommonsRepository.save(incomingUserCommons);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java]: instead of working with "incomingUserCommons", you'd work with just the `userCommons` that you looked up in the database.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java [hunk 3]
```diff
@@ -57,4 +69,59 @@ public UserCommons getUserCommonsById(
...
+      @ApiOperation(value = "Sell a cow, totalWealth updated")
+  @PreAuthorize("hasRole('ROLE_USER')")
+  @PutMapping("/sell")
+  public ResponseEntity<String> putUserCommonsByIdSell(
+          @ApiParam("commonsId") @RequestParam Long commonsId, 
+            @RequestBody @Valid UserCommons incomingUserCommons) throws JsonProcessingException {
+        User u = getCurrentUser().getUser();
+        Long userId = u.getId();
+
+        Commons commons = commonsRepository.findById(commonsId).orElseThrow(
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java]: Same changes here.

**Review by pconrad — CHANGES_REQUESTED** [review]
Looks like this is not set up correctly.    The userCommons should come from a database lookup; it should not be supplied by the user.

If you have the commonsId, that's all you should need; the currentUser is already known to the backend, and with the commonsId which you are already passing in, you can look up the userCommons.

**Comment by sophiajmoore** [comment]
Bad PR
